nuxeo-drive-client/nxdrive/engine/dao/sqlite.py

```diff
@@ -1501,3 +1502,44 @@ def remove_filter(self, path):
 
     def _escape(self, _str):
         return _str.replace("'", "''")
+
+    def _create_users_table(self, cursor, force=False):
+        '''
+            Create a new table 'Users' to store the mapping between user-id and user's full name
+        '''
+        if force:
+            statement = ''
+        else:
+            statement = 'if not exists '
+        # Cannot force UNIQUE for local_path as duplicate can have virtual the same path until they are resolved by Processor
+        # Should improve that
+        cursor.execute("CREATE TABLE " + statement + "Users(user_id VARCHAR NOT NULL PRIMARY KEY, first_name VARCHAR, last_name VARCHAR, last_refreshed TIMESTAMP);")
+
+    def insert_user_info(self, userid, firstName, lastName):
+        log.trace("Inserting user to database: %r" % userid)
+        self._lock.acquire()
+        try:
```

Comment:
catch db exceptions here, log them if necessary. Note that some exception, e.g. db locked is handled internally by the custom exception, to retry it execution (up to 3 times, I think).
I don't think other db exception need to be propagated outside, as the caller cannot take any corrective action.